### PR TITLE
fix(chat): auto-dispatch queued messages on background workspaces

### LINKED ISSUE #851: fix(chat): queued messages only auto-fire on the focused workspace — should fire on background workspaces too
## Symptom

If I'm on workspace A and have queued one or more messages on workspace B, the moment B's agent finishes its turn the queue **should** auto-fire. It doesn't. Nothing happens until I click into B — at which point the queue drains. This breaks the whole point of \"queue and walk away\": I can't actually walk away to another workspace.

## Root cause

The auto-dispatch effect lives inside the singleton `ChatPanel` component, which is only rendered for the currently-selected workspace:

- `src/ui/src/components/layout/AppLayout.tsx:157` — `<ChatPanel />` is mounted once and re-rendered per selected workspace; there is no per-workspace mounted panel hierarchy.
- `src/ui/src/components/chat/ChatPanel.tsx:983-1016` — the `useEffect` that calls `shouldAutoDispatchQueuedMessage(...)` reads `activeSessionId`, `isRunning`, and `queuedMessages[activeSessionId]`. All three resolve to the *focused* workspace.
- `src/ui/src/components/chat/ChatPanel.tsx:386-391` — `activeSessionStatus` / `isRunning` are derived from the *selected* workspace's session, not from the session that owns each queued message.

The data side is fine. The queue store is keyed by session id (not workspace id):
- `src/ui/src/stores/slices/agentInteractionSlice.ts:146` — `queuedMessages: Record<string, QueuedMessage[]>`.

So the queue itself survives workspace switches and is fully addressable for every session — it's only the *driver* (the effect that watches \"agent went idle for session X → flush X's queue\") that is panel-scoped.

`handleSend` itself (`ChatPanel.tsx:1191`) is also local to the focused panel: it persists the user message to the store, calls either `sendRemoteCommand(... 'send_chat_message', ...)` for remote-attached workspaces or `sendChatMessage(...)` for local, then updates UI state. To dispatch on behalf of a background workspace, we need the same routing without the panel being mounted.

## Repro

1. Open two workspaces A and B with active agent sessions.
2. In B, send a long-running prompt so the agent is busy.
3. While B is still running, type a follow-up into B's input and click \"Queue\" (or hit the queue shortcut). The Queued badge shows `1`.
4. Switch to workspace A.
5. Wait for B's agent to finish (you can verify via the sidebar status pill — B goes Running → Idle).
6. **Expected:** B's queued message fires automatically; the next time you visit B the new turn is already in progress / done.
7. **Actual:** B sits Idle with the queued message still queued. Clicking into B is what finally drains the queue.

## Proposal

Two reasonable shapes; I lean (1) for scope, with (2) as a follow-up if we want queues to survive UI crash / restart.

### Option 1 — Lift auto-dispatch to a top-level hook (frontend-only)

Move the auto-dispatch effect out of `ChatPanel` into a hook owned at app-shell scope (e.g. `hooks/useQueuedMessageAutoDispatch.ts` mounted in `App.tsx` or `AppLayout.tsx`). The hook:

- Subscribes to the entire `queuedMessages` map and to all workspaces' `agent_status` (the existing per-workspace status already streams into the store).
- For each `sessionId` with a non-empty queue, evaluates the same `shouldAutoDispatchQueuedMessage(...)` predicate using *that session's* `isRunning`, instead of only the focused session's.
- Dispatches via a session-keyed send helper.

The blocker is that `handleSend` is currently a closure over the focused panel's state. Extract a `dispatchQueuedMessage(sessionId, queuedMessage)` helper into `services/` (or a thin store action) that does only the cross-workspace-safe subset:

- Resolve workspace from session id (already in the store).
- Persist the user message to the chat store for that session.
- Route to `sendRemoteCommand(remote_connection_id, 'send_chat_message', ...)` if the workspace is remote-attached, else `sendChatMessage(...)`.
- Apply mention files / attachments / plan-mode + thinking flags from the queued message payload (already serialized on the queue entry).

Slash-command parsing (`ChatPanel.tsx:1213+`) and prompt-expansion would need to either run inside `dispatchQueuedMessage` too, or be normalized at queue time so the queued entry is already in dispatch-ready form. Preferring the latter — we already do mention-file resolution at queue time.

### Option 2 — Move the queue + auto-flush to the backend (follow-up)

Persist the queue server-side (per `chat_session_id`), and have the agent supervisor flush the next queued entry as soon as its turn returns to Idle. UI becomes a thin view over the server queue. Survives webview crashes, app restart, remote-disconnect. Bigger lift; out of scope for the bug fix but worth a TDD if we want this property.

## Acceptance

- Queue a message on background workspace B while A is focused; when B's agent goes Idle, the queued message dispatches automatically without B regaining focus.
- The queued-message Queued badge on B's sidebar entry decrements (or disappears) without focus change.
- Existing guards still hold: `queuedMessageAutoDispatchPaused`, `isEditingQueuedMessage`, `autoDispatchQueuedIdRef`, `isSteeringQueued` continue to gate dispatch correctly per session.
- Remote-attached workspaces dispatch via the remote bridge (`send_chat_message` over the connection) — not via the local `sendChatMessage` Tauri command.
- No double-dispatch when the user *does* switch back to B mid-flush (the lifted hook is the single owner; the old in-`ChatPanel` effect is removed).
- Vitest: add a test that simulates two sessions where session B becomes idle with a non-empty queue while `activeSessionId === \"A\"` — assert the dispatcher fires for B.

## Out of scope

- Persisting queued messages across app launches (Option 2 territory).
- Changing the queue UX itself (edit-in-place popover, pause toggle, etc).
- Cross-harness migration prelude (`commands/chat/send.rs:1561`) is a separate \"queued prelude\" concept — not the user-facing queue.

## Code pointers

- `src/ui/src/components/chat/ChatPanel.tsx:983-1016` — the panel-scoped effect to extract.
- `src/ui/src/components/chat/queuedMessageEditing.ts:37-55` — `shouldAutoDispatchQueuedMessage`, reusable as-is.
- `src/ui/src/components/chat/ChatPanel.tsx:1191-1558` — `handleSend` / `handleSendRef.current = handleSend`; the cross-workspace `dispatchQueuedMessage` helper should be carved out of this.
- `src/ui/src/stores/slices/agentInteractionSlice.ts:146-340` — queue store, already keyed by session id.
- `src/ui/src/components/layout/AppLayout.tsx:157` — single ChatPanel mount point; the new hook lives one level up.

### PR DESCRIPTION
**Pull request by jamesbrink**
## Summary

- Lift queued-message auto-dispatch out of the singleton focused `ChatPanel` and into an app-shell hook that scans every session queue.
- Extract a session-keyed chat dispatch helper so background queues can send through the same local or remote path without needing their panel mounted.
- Track queued-message editing state in the store so paused/editing guards still prevent auto-drain for the affected session.
- Add Vitest coverage for a queued background workspace going Running -> Idle while another workspace remains focused, plus remote-routing and paused/editing guards.

## Root Cause

Issue #851 happened because the queue itself was already keyed by chat session, but the auto-dispatch effect lived inside `ChatPanel`. `ChatPanel` is mounted once for the selected workspace, so the effect only observed `activeSessionId`, `isRunning`, and the queued messages for the focused workspace. Background sessions could become idle with queued work ready, but no mounted effect was watching that session until the user clicked back into it.

## User Impact

Queued follow-up prompts now fire as soon as their owning session becomes idle, even when the user is working in another workspace. Remote-attached workspaces keep using `send_chat_message` over the remote bridge, and local workspaces keep using the local `sendChatMessage` command.

## Validation

- `cd src/ui && bunx tsc -b --pretty false`
- `cd src/ui && bun run test -- useQueuedMessageAutoDispatch`
- `cd src/ui && bun run lint -- src/hooks/useQueuedMessageAutoDispatch.ts src/hooks/useQueuedMessageAutoDispatch.test.tsx src/components/chat/chatMessageDispatch.ts src/components/chat/ChatPanel.tsx src/components/layout/AppLayout.tsx src/stores/slices/agentInteractionSlice.ts` (0 errors; existing repo warnings only)
- `git diff --check`

Closes #851

### CODE REVIEW

File: src/ui/src/components/chat/ChatPanel.tsx [hunk 1]
```diff
@@ -1444,127 +1384,36 @@ export function ChatPanel() {
...
-      }
+      await dispatchChatMessage({
+        sessionId,
+        content: trimmed,
+        mentionedFiles: mentionedFilesArray,
+        attachments,
+      });
     } catch (e) {
       const errMsg = String(e);
       console.error("sendChatMessage failed:", errMsg);
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/chat/ChatPanel.tsx]: Valid, fixed in 553dd636. The catch block now logs dispatchChatMessage failed so local and remote send failures are labeled accurately.

File: src/ui/src/hooks/useQueuedMessageAutoDispatch.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  const sessionsByWorkspace = useAppStore((s) => s.sessionsByWorkspace);
+  const removeQueuedMessage = useAppStore((s) => s.removeQueuedMessage);
+  const autoDispatchQueuedIdsRef = useRef<Record<string, string>>({});
+
+  useEffect(() => {
+    const state = useAppStore.getState();
+    for (const [sessionId, messages] of Object.entries(queuedMessages)) {
+      const nextQueuedMessage = messages[0];
+      const isRunning = sessionIsRunning(state, sessionId);
+      if (isRunning === null) continue;
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/hooks/useQueuedMessageAutoDispatch.ts]: Valid, fixed in 553dd636. The hook now clears queued state when a queued session no longer exists, and removeChatSession also cleans queued messages plus paused/editing/steering flags so archived/removed sessions do not leave stale queues behind.

File: src/ui/src/hooks/useQueuedMessageAutoDispatch.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      if (!shouldAutoDispatchQueuedMessage({
+        isSteeringQueued: false,
+        isRunning,
+        activeSessionId: sessionId,
+        hasNextQueuedMessage: !!nextQueuedMessage,
+        isEditingQueuedMessage: editing[sessionId] === true,
+        isAutoDispatchPaused: paused[sessionId] === true,
+        autoDispatchQueuedId: autoDispatchQueuedIdsRef.current[sessionId] ?? null,
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/hooks/useQueuedMessageAutoDispatch.ts:41]: Valid, fixed in 553dd636. I added per-session queued-message steering state, wired both skip-queue and queued-message steer paths to set/clear it, and now the app-shell auto-dispatch hook uses that session-keyed guard instead of a hard-coded false value.
